src/ArtifactBuilds.jsx

```diff
@@ -1,42 +1,60 @@
-import { v4 as uuid } from "uuid";
+import _ from "lodash";
 
-import React from "react";
+import PropTypes from "prop-types";
 import { Button, Grid } from "@mui/material";
 
 import ArtifactBuild from "./ArtifactBuild/ArtifactBuild";
-import _ from "lodash";
-
-class ArtifactBuilds extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { builds: [] };
-  }
-
-  addBuild() {
-    const builds = this.state.builds.concat([{ id: uuid() }]);
-    this.setState({ builds });
-  }
 
-  render() {
-    return (
-      <Grid container>
-        <Grid item xs={12}>
-          <Grid container spacing={2}>
-            {_.map(this.state.builds, (build) => {
-              return (
-                <Grid item key={build.id} xs={12} sm={6} md={4} lg={3} xl={2}>
-                  <ArtifactBuild build={build} />
-                </Grid>
-              );
-            })}
-          </Grid>
-        </Grid>
-        <Grid item xs={12}>
-          <Button onClick={this.addBuild.bind(this)}>Add Build</Button>
+function ArtifactBuilds(props) {
+  return (
+    <Grid container>
+      <Grid item xs={12}>
+        <Grid container spacing={2}>
+          {_.map(props.builds, (build) => {
+            return (
+              <Grid item key={build.id} xs={12} sm={6} md={4} lg={3} xl={2}>
+                <ArtifactBuild
+                  addRestriction={() => props.addRestriction(build.id)}
+                  addSetRestriction={() => props.addSetRestriction(build.id)}
+                  build={build}
+                  onCharacterChange={(character) =>
+                    props.onCharacterChange(build.id, character)
+                  }
+                  onRestrictionRemoval={(restrictionId) =>
+                    props.onRestrictionRemoval(build.id, restrictionId)
```

Comment:
I gotta do this thing, where I basically change the method signature to embed the build ID. This seems weird to me too.

Reply:
Alternatively, you can just pass the restrictionId along as a prop and then the caller can provide it directly